x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/common/H3CartesianUtil.java

```diff
@@ -170,6 +187,34 @@ public static LatLonGeometry getLatLonGeometry(long h3) {
         return new H3CartesianGeometry(h3);
     }
 
+    /** Return the {@link Geometry} representing the provided H3 bin */
+    public static Geometry getNormalizeGeometry(long h3) {
+        final double[][] cached = CACHED_H3.get(h3);
+        final double[] xs;
+        final double[] ys;
+        if (cached != null) {
+            xs = cached[0].clone();
+            ys = cached[1].clone();
+        } else {
+            final CellBoundary boundary = H3.h3ToGeoBoundary(h3);
+            final int numPoints = numPoints(h3, boundary);
+            xs = new double[numPoints];
+            ys = new double[numPoints];
+            computePoints(h3, boundary, xs, ys);
+        }
+        final Polygon polygon = new Polygon(new LinearRing(xs, ys));
+        if (isPolar(h3) || GeometryNormalizer.needsNormalize(Orientation.CCW, polygon) == false) {
+            return polygon;
+        }
+        final Geometry geometry = GeometryNormalizer.apply(Orientation.CCW, polygon);
+        if (geometry instanceof MultiPolygon) {
+            return geometry;
+        }
+        // we shouldn't be here but one of the polygons crossing the dateline fails
```

Comment:
Curious, would like to know more about this?

Reply:
I open https://github.com/elastic/elasticsearch/issues/92949 which I think it captures the issue